core/src/main/java/org/apache/iceberg/MetadataTableUtils.java

```diff
@@ -32,9 +32,16 @@ public static boolean hasMetadataTableName(TableIdentifier identifier) {
   public static Table createMetadataTableInstance(Table table, MetadataTableType type) {
     if (table instanceof BaseTable) {
       return createMetadataTableInstance(table, metadataTableName(table.name(), type), type);
+    } else if (table instanceof HasTableOperations) {
+      return createMetadataTableInstance(
+          ((HasTableOperations) table).operations(),
+          table.name(),
+          metadataTableName(table.name(), type),
+          type);
     } else {
       throw new IllegalArgumentException(
-          String.format("Cannot create metadata table for table %s: not a base table", table));
+          String.format(
```

Comment:
'table is not a base table or does not have table operations'?

Reply:
Changed as suggested